Sources/AWSLambdaDeploymentDescriptorGenerator/JSONSChemaReader.swift

```
@@ -44,116 +84,329 @@ enum JSONPrimitiveType: Decodable, Equatable {
+    case reference(String)
+    
     
     enum CodingKeys: String, CodingKey {
-        case anyOf
-        case allOf
+        case anyOf = "anyOf"
+        case allOf = "allOf"
+        case ref = "$ref"
     }
     
-    init(from decoder: any Decoder) throws {
+    
+    init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
-        var allKeys = ArraySlice(container.allKeys)
-        if let onlyKey = allKeys.popFirst(), allKeys.isEmpty  {
-            // there is an anyOf or allOf key
-            switch onlyKey {
-            case .allOf: fatalError("not yet implemented")
-            case .anyOf:
-                let value = try container.decode(Array<JSONType>.self, forKey: .anyOf)
-                self = .anyOf(value)
+        do {
+           
+            var allKeys = ArraySlice(container.allKeys)
+            
+            
+            if let onlyKey = allKeys.popFirst(), allKeys.isEmpty {
+                
+                switch onlyKey {
+                case .anyOf:
+                    let value = try container.decode([JSONType].self, forKey: .anyOf)
+                    self = .anyOf(value)
+                    
+                case .allOf:
+                    let values = try container.decode([JSONType].self, forKey: .allOf)
+                    for value in values {
+                        if !(value is JSONType) {
+                            throw DecodingError.typeMismatch(JSONUnionType.self, DecodingError.Context(codingPath: container.codingPath, debugDescription: "➡️ Unexpected element type in allOf", underlyingError: nil))
+                        }
+                    }
+                    
+                    self = .allOf(values)
+                case .ref:
+                    let refString = try container.decode(String.self, forKey: .ref)
+                    self = .reference(refString)
+                }
+            } else {
+                do {
```

not necessary to catch the error here, just use try and let errors propagate up to the stack